communication/lib/mbedtls/library/debug.c

```diff
@@ -70,7 +70,7 @@ static inline void debug_send_line( const mbedtls_ssl_context *ssl, int level,
      */
 #if defined(MBEDTLS_THREADING_C)
     char idstr[20 + DEBUG_BUF_SIZE]; /* 0x + 16 nibbles + ': ' */
-    mbedtls_snprintf( idstr, sizeof( idstr ), "%p: %s", ssl, str );
+    mbedtls_snprintf( idstr, sizeof( idstr ), "%p: %s\r\n", ssl, str );
```

Comment:
I did originally do that via `debug_send_line`, but there are some cases where it's not appropriate, like a buffer dump.  So the higher level API needed modification instead.  The above change to line 73 I believe was residual testing that should have been removed (MBEDTLS_THREADING_C is not currently defined).  If we add `\r\n` to `my_debug` most of the log messages would end with `\n\r\n` adding a bunch of space to the log output.  I agree we should try to minimize changes to 3rd party libraries, but we also need to have effective logs.  I think these changes should merge with upstream MBEDTLS library changes without conflict.  I'll revert the change on line 73.